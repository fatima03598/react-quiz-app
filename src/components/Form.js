import React, { Component } from 'react';
import {Link} from 'react-router-dom';
class Form extends Component {
    constructor(){
        super()
        this.state={

                userNames: {
                    1: "",
                    2: "",
                    3: "",
                    4: ""
                },
                forms: {
                    difficulty: "easy",
                    playersNumber: 1,
                    category: "9"
                },
                data: null,	
            
        }
      

    }

  

    updateState = (e) => {
        e.preventDefault();
        const obj = e.target.name
        const name = e.target.value
        this.setState({ forms: { ...this.state.forms, [obj]: name}})
        // this.setState({[obj]: e.target.value})
    }

  
    render() {
        console.log(this.state);
        return (
            <div>
                <form>
                    <label>
                        How many players:
                        <select 
                            onChange={this.updateState} 
                            value={this.state.forms.playersNumber} 
                            name="playersNumber">
                                <option value="1">1</option>
                                <option value="2">2</option>
                                <option value="3">3</option>
                                <option value="4">4</option>
                        </select>
                    </label>
                    <label>
                        Choose a category:
                        <select 
                            onChange={this.updateState} 
                            value={this.state.category}
                            name="category"
                            >
                                <option value="any">Any Category</option>
                                <option value="9">General Knowledge</option>
                                <option value="10">Entertainment: Books</option>
                                <option value="11">Entertainment: Film</option>
                                <option value="12">Entertainment: Music</option>
                                <option value="13">Entertainment: Musicals &amp; Theatres</option>
                                <option value="14">Entertainment: Television</option>
                                <option value="15">Entertainment: Video Games</option>
                                <option value="16">Entertainment: Board Games</option>
                                <option value="17">Science &amp; Nature</option>
                                <option value="18">Science: Computers</option>
                                <option value="19">Science: Mathematics</option>
                                <option value="20">Mythology</option>
                                <option value="21">Sports</option>
                                <option value="22">Geography</option>
                                <option value="23">History</option>
                                <option value="24">Politics</option>
                                <option value="25">Art</option>
                                <option value="26">Celebrities</option>
                                <option value="27">Animals</option>
                                <option value="28">Vehicles</option>
                                <option value="29">Entertainment: Comics</option>
                                <option value="30">Science: Gadgets</option>
                                <option value="31">Entertainment: Japanese Anime &amp; Manga</option>
                                <option value="32">Entertainment: Cartoon &amp; Animations</option>
                        </select>
                    </label>
                    <label>
                        How difficult:
                        <select 
                            onChange={this.updateState} 
                            value={this.state.difficulty}
                            name="difficulty">
                                <option value="easy">Easy</option>
                                <option value="medium">Medium</option>
                                <option value="hard">Hard</option>
                        </select>
                    </label>
                    <Link to={`/gamepage/${this.state.forms.playersNumber}/${this.state.forms.difficulty}/${this.state.forms.category}`}>
                    <button>Submit and Start the Game</button>
                    </Link>
                    
                </form>
            </div>
        );
    }
}

export default Form;