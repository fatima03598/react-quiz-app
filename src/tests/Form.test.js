import React from 'react'
import { BrowserRouter, Link } from 'react-router-dom';
import { shallow, mount } from 'enzyme'
import Form from '../components/Form'

describe('<Form />', () => {

    it('should render a form', () => {
        const wrapper = shallow(<Form />); 
        expect(wrapper.exists('form')).toEqual(true);
    } )


    
    it('should have 3 select options', () => {
        const wrapper = shallow(<Form />); 
        expect(wrapper.find('select').length).toEqual(3);
    } )



    it('On change the value should change', () => {
        const wrapper = mount(<BrowserRouter><Form /></BrowserRouter>); 
        wrapper.find('select').at(0).simulate('change', { target: { name: 'playersNumber', value: 3 } });
        expect(wrapper.find('select').at(0).prop('value')).toEqual(3);

    })




  })