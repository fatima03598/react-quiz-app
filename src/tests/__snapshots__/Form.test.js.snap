// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> matches the snapshot 1`] = `
<div
  className="form-container"
>
  <form
    className="quiz-form"
  >
    <label
      className="game-start-input"
    >
      Enter the number of players:
      <select
        name="playersNumber"
        onChange={[Function]}
        value={1}
      >
        <option
          value="1"
        >
          1
        </option>
        <option
          value="2"
        >
          2
        </option>
        <option
          value="3"
        >
          3
        </option>
        <option
          value="4"
        >
          4
        </option>
      </select>
    </label>
    <label
      className="game-start-input"
    >
      Choose your quiz category:
      <select
        name="category"
        onChange={[Function]}
      >
        <option
          value="9"
        >
          General Knowledge
        </option>
        <option
          value="27"
        >
          Animals
        </option>
        <option
          value="25"
        >
          Art
        </option>
        <option
          value="26"
        >
          Celebrities
        </option>
        <option
          value="16"
        >
          Entertainment: Board Games
        </option>
        <option
          value="10"
        >
          Entertainment: Books
        </option>
        <option
          value="32"
        >
          Entertainment: Cartoon & Animations
        </option>
        <option
          value="29"
        >
          Entertainment: Comics
        </option>
        <option
          value="11"
        >
          Entertainment: Film
        </option>
        <option
          value="31"
        >
          Entertainment: Japanese Anime & Manga
        </option>
        <option
          value="12"
        >
          Entertainment: Music
        </option>
        <option
          value="13"
        >
          Entertainment: Musicals & Theatres
        </option>
        <option
          value="14"
        >
          Entertainment: Television
        </option>
        <option
          value="15"
        >
          Entertainment: Video Games
        </option>
        <option
          value="22"
        >
          Geography
        </option>
        <option
          value="23"
        >
          History
        </option>
        <option
          value="20"
        >
          Mythology
        </option>
        <option
          value="24"
        >
          Politics
        </option>
        <option
          value="18"
        >
          Science: Computers
        </option>
        <option
          value="30"
        >
          Science: Gadgets
        </option>
        <option
          value="19"
        >
          Science: Mathematics
        </option>
        <option
          value="17"
        >
          Science & Nature
        </option>
        <option
          value="21"
        >
          Sports âš½
        </option>
        <option
          value="28"
        >
          Vehicles
        </option>
      </select>
    </label>
    <label
      className="game-start-input"
    >
      Choose the difficulty level:
      <select
        name="difficulty"
        onChange={[Function]}
      >
        <option
          value="easy"
        >
          Easy ðŸ¤‘
        </option>
        <option
          value="medium"
        >
          Medium ðŸ¤‘
        </option>
        <option
          value="hard"
        >
          Hard ðŸ¤‘
        </option>
      </select>
    </label>
    <a
      href="/gamepage/1/easy/9"
      onClick={[Function]}
    >
      <button
        className="start-game-button"
      >
        Start Game
      </button>
    </a>
  </form>
</div>
`;
